task/mongo-optimization-task.js

```diff
@@ -241,23 +212,23 @@ async function task_3_1(db) {
             }
         },
         {
-            "$lookup" : {
-                "from" : "clientCriteria",
-                "localField" : "criteria_value",
-                "foreignField" : "value",
-                "as" : "criteria"
-            }
-        },
-        {
-            "$unwind" : "$criteria"
-        },
-        {
-            "$unwind" : "$criteria.versions"
-        },
-        {
-            "$match" : {
-                "criteria.versions.initiativeId" : ObjectId("58af4da0b310d92314627290")
-            }
+           "$lookup" : {
+               "from" : "clientCriteria",
+                    let: { "opportunities_value": "$criteria_value"},
+                    pipeline: [
+                    {
+                        $match : {
+                            "versions.initiativeId" : ObjectId("58af4da0b310d92314627290")
+                        }
+                    },                   
```

Comment:
you can make it on one `$match`. Also, projection and `$limit` can help you here